service/assessment/assessment.go

```diff
@@ -53,6 +56,16 @@
 	log = logrus.WithField("component", "assessment")
 }
 
+const (
+	// EvictionTime is the time after which an entry in the metric configuration is invalid
+	EvictionTime = time.Minute * 5
```

Comment:
As you mentioned in the PR Description, an event trigger (sth. like `onMetricConfigurationChanged()`)would be better but for now it should be ok, even if 5 minutes sounds too low for me since metric configurations shouldn't change that often. (Also the eviction time of 5 minutes is as long as the discovery interval^^)